src/profile-logic/process-profile.js

```diff
@@ -732,6 +733,79 @@ function _processCounters(
   );
 }
 
+/**
+ * Converts the given visual progress into the processed format.
+ */
+function _processVisualProgress(
+  geckoProfile: GeckoProfile,
+  stableThreadList: Thread[]
+): Counter[] {
+  const visualProgressString = geckoProfile.meta.visualMetrics
+    ? geckoProfile.meta.visualMetrics.VisualProgress
+    : '';
+  const mainThread = geckoProfile.threads.find(
+    thread => thread.name === 'GeckoMain'
+  );
+
+  if (!mainThread || !visualProgressString) {
+    // Visual Progress or a main thread weren't found, bail out, and return an empty array.
+    return [];
+  }
+
+  // The gecko profile's process don't map to the final thread list. Use the stable
+  // thread list to look up the thread index for the main thread in this profile.
+  const mainThreadIndex = stableThreadList.findIndex(
+    thread => thread.name === 'GeckoMain' && thread.pid === mainThread.pid
+  );
+
+  if (mainThreadIndex === -1) {
+    throw new Error(
+      'Unable to find the main thread in the stable thread list. This means that the ' +
+        'logic in the _processCounters function is wrong.'
+    );
+  }
+
+  const samples = {
+    count: [],
+    number: [],
+    time: [],
+    length: 0,
+  };
+
+  // Need to offset sample time by thread range start for time to show correctly
+  // in profiler
+  const timeOffset = getMinRangeStartIncludingAllThreads(
+    stableThreadList,
+    geckoProfile
+  );
+
+  visualProgressString.split(', ').forEach((visualProgress, index) => {
```

Comment:
Please document what this `visualProgressString` content looks like.

Reply:
After some recent changes to browsertime and changes to my approach, visual progress data no longer needs to be processed and is ready to consume.